parser-typechecker/src/Unison/DeclPrinter.hs

```diff
@@ -144,9 +144,9 @@ fieldNames env r name dd = case DD.constructors dd of
     fieldNames = Map.fromList
       [ (r, f) | (r, n) <- names
                , typename <- pure (HQ.toString name)
-               , typename `isPrefixOf` n
-               -- drop the typename and the following '.'
-               , rest <- pure $ drop (length typename + 1) n
+               , rest <- pure $ if typename `isPrefixOf` n then 
+                                drop (length typename + 1) n 
+                                else n
```

Comment:
I'm not sure this fix is going to do it for all cases, since the PPE that's passed in is going to be returning suffixed names. The suffixed name could actually be `set` or `modify` if this type is the only record type in the namespace.

Reply:
I don't see it. The only things in this `names` map are the accessors generated within this function. The passed-in PPE is only used for looking up the names for those. I must be missing something.

Reply:
The PPE that's passed in could report that the name for one of the record accessors is actually just `modify` (because just `modify` is a unique suffix in the namespace).

Another case I thought of is that one of the record accessors could have a name of `base.v12.foo.bar.Blah.modify` (because the namespace has a `base.v11.foo.bar.Blah` type as well). I don't think this PR (or what's in trunk) will correctly handle this case.